File: Add Code Here/C++/Valid_Parenthese.cpp
class Solution {
public:
    bool isValid(string s) {
        int length=s.size();
        vector<char> brackets;
        for(int i=0;i<length;i++)
        {
            if(brackets.empty())
            {
                brackets.push_back(s[i]);
            }
            else if( brackets.back() == '(' && s[i] == ')' || brackets.back() == '[' && s[i] == ']' || brackets.back() == '{' && s[i] == '}' )
            {
                brackets.pop_back();
            }
            else
            {
                brackets.push_back(s[i]);
            }
        }
        if(brackets.empty())
        {
            return true;
        }
            return false;
        
        
        
    }
};

File: Add Code Here/PYTHON/ValidParenthese.py
class Solution:
    def isValid(self, s: str) -> bool:
        if len(s) % 2 != 0:
            return False
        dict = {'(' : ')', '[' : ']', '{' : '}'}
        stack = []
        for iin s:
            if i in dict.keys():
                stack.append(i)
            else:
                if stack == []:
                    return False
                a = stack.pop()
                if i!= dict[a]:
                    return False
        return stack == []
